dotnet/src/Connectors/Connectors.Memory.Weaviate/WeaviateMemoryStore.cs

```
@@ -0,0 +1,516 @@
+            metadata: ToMetadata(weaviateObject));
+
+        this._log.LogTrace("Vector found with key: {0}", key);
+
+        return record;
+    }
+
+    /// <inheritdoc />
+    public async IAsyncEnumerable<MemoryRecord> GetBatchAsync(string collectionName, IEnumerable<string> keys, bool withEmbeddings = false,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        foreach (string? key in keys)
+        {
+            MemoryRecord? record = await this.GetAsync(collectionName, key, withEmbeddings, cancellationToken).ConfigureAwait(false);
+            if (record != null)
+            {
+                yield return record;
+            }
+            else
+            {
+                this._log.LogWarning("Unable to locate object with id: {0}", key);
+            }
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task RemoveAsync(string collectionName, string key, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNullOrEmpty(collectionName, "Collection name is empty");
+        Verify.NotNull(key, "Key is NULL");
+
+        string className = ToWeaviateFriendlyClassName(collectionName);
+        this._log.LogTrace("Deleting vector with key: {0}, from collection {1}, with class name: {2}:", key, collectionName, className);
+
+        DeleteObjectRequest requestBuilder = new()
+        {
+            Class = className,
+            Id = key
+        };
+        using HttpRequestMessage request = requestBuilder.Build();
+
+        (HttpResponseMessage response, string _) = await this.ExecuteHttpRequestAsync(request, cancellationToken).ConfigureAwait(false);
+
+        try
+        {
+            response.EnsureSuccessStatusCode();
+            this._log.LogTrace("Vector deleted");
+        }
+        catch (HttpRequestException e)
+        {
+            this._log.LogError(e, "Vector delete request failed: {0}", e.Message);
+        }
+    }
```

We should re-throw `WeaviateMemoryException` if removal failed, otherwise we won't be able to know whether vector was deleted or not.

---

In that case this needs to be changed in the QDrant connector too as it is swallowing exceptions and writing to the logs.